// highcharts-angular/fesm2015/highcharts-angular.js
import { __decorate, __metadata } from 'tslib';
import { EventEmitter, ElementRef, NgZone, Input, Output, Component, NgModule } from '@angular/core';
import 'highcharts';

let HighchartsChartComponent = class HighchartsChartComponent {
    constructor(el, _zone // #75
    ) {
        this.el = el;
        this._zone = _zone;
        this.updateChange = new EventEmitter(true);
        this.chartInstance = new EventEmitter(); // #26
    }
    ngOnChanges(changes) {
        const update = changes.update && changes.update.currentValue;
        if (changes.options || update) {
            this.wrappedUpdateOrCreateChart();
            if (update) {
                this.updateChange.emit(false); // clear the flag after update
            }
        }
    }
    wrappedUpdateOrCreateChart() {
        if (this.runOutsideAngular) {
            this._zone.runOutsideAngular(() => {
                this.updateOrCreateChart();
            });
        }
        else {
            this.updateOrCreateChart();
        }
    }
    updateOrCreateChart() {
        if (this.chart && this.chart.update) {
            this.chart.update(this.options, true, this.oneToOne || false);
        }
        else {
            this.chart = this.Highcharts[this.constructorType || 'chart'](this.el.nativeElement, this.options, this.callbackFunction || null);
            // emit chart instance on init
            this.chartInstance.emit(this.chart);
        }
    }
    ngOnDestroy() {
        if (this.chart) { // #56
            this.chart.destroy();
            this.chart = null;
        }
    }
};
HighchartsChartComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: NgZone // #75
     }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], HighchartsChartComponent.prototype, "Highcharts", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], HighchartsChartComponent.prototype, "constructorType", void 0);
__decorate([
    Input(),
    __metadata("design:type", Function)
], HighchartsChartComponent.prototype, "callbackFunction", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], HighchartsChartComponent.prototype, "oneToOne", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], HighchartsChartComponent.prototype, "runOutsideAngular", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], HighchartsChartComponent.prototype, "options", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], HighchartsChartComponent.prototype, "update", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], HighchartsChartComponent.prototype, "updateChange", void 0);
__decorate([
    Output(),
    __metadata("design:type", Object)
], HighchartsChartComponent.prototype, "chartInstance", void 0);
HighchartsChartComponent = __decorate([
    Component({
        selector: 'highcharts-chart',
        template: ''
    }),
    __metadata("design:paramtypes", [ElementRef,
        NgZone // #75
    ])
], HighchartsChartComponent);

let HighchartsChartModule = class HighchartsChartModule {
};
HighchartsChartModule = __decorate([
    NgModule({
        declarations: [HighchartsChartComponent],
        exports: [HighchartsChartComponent]
    })
], HighchartsChartModule);

/*
 * Public API Surface of highcharts-angular
 */

/**
 * Generated bundle index. Do not edit.
 */

export { HighchartsChartComponent, HighchartsChartModule };
